egs2/swbd/slu1/local/turn_take_utils.py

```diff
@@ -0,0 +1,237 @@
+            if word_val != "":
+                line1_better = [line1[0], line1[1], line1[2], word_val]
+                line_file_filtered_A.append(line1_better)
+        return line_file_filtered_A, []
+    for val in backchannel_dict[file_name]:
+        while count < len(line_file_unfiltered):
+            line1 = line_file_unfiltered[count].split()
+            count += 1
+            if (line1[0]) == val:
+                break
+            else:
+                word_val = clean_line(line1[3].strip())
+                if word_val != "":
+                    line1_better = [line1[0], line1[1], line1[2], word_val]
+                    line_file_filtered_A.append(line1_better)
+        if (line1[0]) in backchannel_dict[file_name]:
+            for sub_val in backchannel_dict[file_name][line1[0]]:
+                found_1 = False
+                found_2 = False
+                while count < len(line_file_unfiltered):
+                    if sub_val[0] == float(line1[1]):
+                        found_1 = True
+                    if sub_val[1] == float(line1[2]):
+                        found_2 = True
+                    if found_1 and found_2:
+                        line_backchannel_A.append(line1)
+                        found_1 = False
+                        found_2 = False
+                        break
+                    elif found_1:
+                        line_backchannel_A.append(line1)
+                    elif found_2:
+                        if sub_val[0] != sub_val[1]:
+                            print("error")
+                            import pdb
+
+                            pdb.set_trace()
+                    else:
+                        word_val = clean_line(line1[3].strip())
+                        if word_val != "":
+                            line1_better = [line1[0], line1[1], line1[2], word_val]
+                            line_file_filtered_A.append(line1_better)
+                    line1 = line_file_unfiltered[count].split()
+                    count += 1
+    while count < len(line_file_unfiltered):
+        line1 = line_file_unfiltered[count].split()
+        count += 1
+        word_val = clean_line(line1[3].strip())
+        if word_val != "":
+            line1_better = [line1[0], line1[1], line1[2], word_val]
+            line_file_filtered_A.append(line1_better)
+    return line_file_filtered_A, line_backchannel_A
+
+
+def clean_dialogue(word_file_A, word_file_B, sentence_dict_sort):
+    count_A = 0
+    count_B = 0
+    prev_speaker = "NA"
+    sentence_dict_sort_clean = {}
+    for k in sentence_dict_sort:
+        if "A" == sentence_dict_sort[k][0]:
+            if count_A >= len(word_file_A):
+                continue
+            while float(word_file_A[count_A][2]) <= sentence_dict_sort[k][1]:
+                if word_file_A[count_A][0] != sentence_dict_sort[k][2]:
+                    import pdb
+
+                    pdb.set_trace()
```

Comment:
redundent

Reply:
Thanks for pointing this out! I fixed this now!